plugins/pay.c

```diff
@@ -454,19 +454,38 @@ static const jsmntok_t *find_worst_channel(const char *buf,
 					   const char *fieldname,
 					   u64 final)
 {
-	u64 prev = final, worstval = 0;
-	const jsmntok_t *worst = NULL, *t;
+	u64 prev, worstval = 0;
+	const jsmntok_t *worst = NULL, *t, *t_prev = NULL;
 	size_t i;
 
 	json_for_each_arr(i, t, route) {
 		u64 val;
+		/* If there's only 2 hops, the worst channel must be the first one. */
+		if (route->size <= 2) {
+			worst = t;
+			break;
+		}
```

Comment:
Don't special-case this, I don't think it needs it?

Reply:
Yes...there's no need.